pkg/apis/extensions/types.go

```diff
@@ -82,43 +81,45 @@ type SubresourceReference struct {
 	Subresource string `json:"subresource,omitempty"`
 }
 
-// ResourceConsumption is an object for specifying average resource consumption of a particular resource.
-type ResourceConsumption struct {
-	Resource api.ResourceName  `json:"resource,omitempty"`
-	Quantity resource.Quantity `json:"quantity,omitempty"`
+type Utilization float64
+
+type CPUTargetUtilization struct {
+	// Utilization target defines fraction of the requested CPU that should be utilized/used,
+	// e.g. 0.7 means that 70% of the requested CPU should be in use.
+	UtilizationTarget Utilization `json:"utilizationTarget"`
 }
 
 // HorizontalPodAutoscalerSpec is the specification of a horizontal pod autoscaler.
 type HorizontalPodAutoscalerSpec struct {
 	// ScaleRef is a reference to Scale subresource. HorizontalPodAutoscaler will learn the current resource consumption from its status,
 	// and will set the desired number of pods by modyfying its spec.
-	ScaleRef *SubresourceReference `json:"scaleRef"`
+	ScaleRef SubresourceReference `json:"scaleRef"`
 	// MinReplicas is the lower limit for the number of pods that can be set by the autoscaler.
-	MinReplicas int `json:"minReplicas"`
+	MinReplicas *int `json:"minReplicas"`
 	// MaxReplicas is the upper limit for the number of pods that can be set by the autoscaler. It cannot be smaller than MinReplicas.
 	MaxReplicas int `json:"maxReplicas"`
```

Comment:
+1, i think if out the gate, a user MUST specify a value now since 0 is unsupported.  As a result, making it valid to omit I think can cause confusion or make it too easy in the future for the user to accidentally set the value as 0.  Since scale down to 0 could result in future downtime, I think it should be required to set explicitly.